src/Polysemy/RandomFu.hs

```diff
@@ -0,0 +1,110 @@
+gives @Sem r@ a "MonadRandom" instance (from "random-fu"). This is an Orphan instance.
+-}
+
+module Polysemy.RandomFu
+  (
+    -- * Effect
+    Random (..)
+
+    -- * Actions
+  , sampleRVar
+  , getRandomPrim
+  , sampleDist
+
+  -- * Interpretations
+  , runRandomSource
+  , runRandomIO
+  , runRandomIOPureMT
+  )
+where
+
+import           Polysemy                      
+
+import           Data.IORef                     ( newIORef )
+import qualified Data.Random                   as R
+import qualified Data.Random.Source            as R
+import           Data.Random.Source.Std         (StdRandom(..))
+import qualified Data.Random.Internal.Source   as R
+import qualified Data.Random.Source.PureMT     as R
+import qualified System.Random                 as SR
+import           Control.Monad.IO.Class         ( MonadIO(..) )
+
+------------------------------------------------------------------------------
+{- | An effect capable of sampling from a "random-fu" RVar or generating a
+single random-variate of any type, @t@ with a
+@Data.Random.Prim t@ constructor, currently one of @Word8@, @Word16@,
+@Word32@, @Word64@, @Double@ or N-byte integer.
+-}
+data Random m r where
+  SampleRVar ::  R.RVar t -> Random m t
+  GetRandomPrim :: R.Prim t -> Random m t
+
+makeSem ''Random
+
+------------------------------------------------------------------------------
+-- | use the 'Random` effect to sample from a "random-fu" @Distribution@.
+sampleDist
+  :: (Member Random r, R.Distribution d t) => d t -> Sem r t
+sampleDist = sampleRVar . R.rvar
+
+------------------------------------------------------------------------------
+-- | Run a 'Random' effect using a given 'R.RandomSource'
+runRandomSource
+  :: forall s r a
+   . R.RandomSource (Sem r) s
+  => s
+  -> Sem (Random ': r) a
+  -> Sem r a
+runRandomSource source = interpret f
+ where
+  f :: forall m x . (Random m x -> Sem r x)
+  f r = case r of
+    SampleRVar    rv -> R.runRVar (R.sample rv) source
+    GetRandomPrim pt -> R.runRVar (R.getRandomPrim pt) source
+{-# INLINE runRandomSource #-}
+
+------------------------------------------------------------------------------
+-- | Run a 'Random` effect by using the default "random-fu" 'IO' source
+runRandomIO
+  :: forall r a
+   . MonadIO (Sem r)
+  => Sem (Random ': r) a
+  -> Sem r a
+runRandomIO = interpret f
+ where
+  f :: forall m x . (Random m x -> Sem r x)
+  f r = case r of
+    SampleRVar    rv -> liftIO $ R.sample rv
+    GetRandomPrim pt -> liftIO $ R.getRandomPrim pt
+{-# INLINE runRandomIO #-}
+
+------------------------------------------------------------------------------
+-- | Run in 'IO', using the given 'R.PureMT' source, stored in an 'IORef'
+runRandomIOPureMT
+  :: MonadIO (Sem r)
+  => R.PureMT
+  -> Sem (Random ': r) a
+  -> Sem r a
+runRandomIOPureMT source re =
+  liftIO (newIORef source) >>= flip runRandomSource re
+{-# INLINE runRandomIOPureMT #-}
+
+-- | Orphan instance of 'R.MonadRandom'
+$(R.monadRandom [d|
```

Comment:
Why is this template haskell? 

I'd prefer not introducing an orphan here, but I could be talked into it if I knew why it's necessary.

Reply:
The TH generates several more functions in the instance and the random-fu docs say one should do it this way since the implementation of those might change.
I have a bunch of data-science stuff that uses random-fu via the "MonadRandom" constraint.  So this makes that painless.  There might be other users like that.
I think this might be a general tension between thinking of effects as an app-level thing vs. as a specific utility.  I'm using it more in the latter way, where the effects may need to interact with existing monads/mtl constraints.  So I'm trying to make that easy where the costs are low.